src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets

```diff
@@ -0,0 +1,305 @@
+       ArchivesRoot                 - Relative path for work item payloads to use.  Blob Uris will be made relative to this.
+       
+       **********************************************************************************************************************************
+       Optional Properties:
+       **********************************************************************************************************************************
+       TargetsWindows    - Whether to use Unix or Windows-specific path creation.             Default: False
+       TestListFilename  - File name to use for test list sent to Helix.                      Default: TestList.json
+       OverwriteOnUpload - Whether to overwrite blobs if they already exist on upload.        Default: False
+       ContainerName     - Container name for uploaded files.                                 Default: build-<Random GUID>. 
+       IsOfficial        - Boolean flag for display on Mission Control.                       Default: False:
+       HelixApiEndpoint  - Endpoint to send jobs to.                                          Default: https://helix.dot.net/api/2016-06-28/jobs
+       CloudResultsReadTokenValidDays  - Days that result URLs produced will be readble       Default: 30
+       CloudResultsWriteTokenValidDays - Days that result URLs produced will be write         Default: 4
+       
+       **********************************************************************************************************************************
+       Required ITaskItems:
+       **********************************************************************************************************************************
+       HelixWorkItem         - Zip files to be uploaded as payloads. 
+       
+       **********************************************************************************************************************************
+       Optional ITaskItems:
+       **********************************************************************************************************************************
+       HelixCorrelationPayloadFile - Zip files to be uploaded as correlation payloads (shared by all work items, cached where possible)       
+    -->
+
+  <!-- Main entry point -->
+  <Target Name="HelixCloudBuild"
+          DependsOnTargets="VerifyInputs;PreCloudBuild;ValidateWorkItems;UploadContent;CreateTestListJson" />
+
+  <PropertyGroup>
+    <OverwriteOnUpload Condition="'$(OverwriteOnUpload)' == ''">false</OverwriteOnUpload>
+    <ContainerName Condition="'$(ContainerName)'== ''">build-$([System.Guid]::NewGuid().ToString("N"))</ContainerName>
+    <ContainerName>$(ContainerName.ToLower())</ContainerName>
+    <TestListFilename Condition="'$(TestListFilename)'==''">TestList.json</TestListFilename>
+    <SupplementalPayloadDir Condition="'$(SupplementalPayloadDir)' == ''">$([System.IO.Path]::GetTempPath())$([System.IO.Path]::GetRandomFileName())/SupplementalPayload/</SupplementalPayloadDir>
```

Comment:
I guess I was expecting this to be an input but looking at your usage it does appear that is really scoped directory just for these targets so I'm good with it being a random directory and not having it be an optional parameter. Although it would probably be good to clean it up.